### ISSUE TITLE
dom form reset() don't reset input to default value

### ISSUE THREAD

**Issue by futurist** [lhorie/mithril.js#854]
think below form:

````
m('form', [
    m('input', { value:'init value' }),
    m('input', { type:'reset' }),
    m('select', [ m('option', "A"), m('option', {selected:true}, "B") ] ),
])
````
Issues:
1. `select` don't have initial value set to 'B'
2. When click reset button, the input back to empty, instead of it's initial value.
Is this an issue related to dom render?

**Comment by tivac** [comment]
A form reset button changes the DOM out from under Mithril, so it doesn't know that things have changed and is (properly) not redrawing anything.

**Comment by futurist** [comment]
should render the dom according to standard behavior,  form should have initial value for input.

**Comment by barneycarroll** [comment]
tl; dr: you can't use both Mithril _and_ high level form controls to control state - it's one or the other. In Mithril, you should stay away from native functionality that assumes total state control and write your own model functionality.

The problem here is that whereas in plain old static HTML, the page receives a DOM and forms control the their own state according to native logic. When you use Mithril, Mithril creates and manages the DOM based on models you define. In other words: in HTML, the form is the model, the controller and the view all in one. In Mithril, it's a view representation of a model.

In HTML, a reset button remembers the initial state of the form and resets it. In Mithril, the view is stateless. You define the state of the form model and how it's controlled, and Mithril renders it. These methods of doing things are ultimately in contradiction with each-other. Forms allow a limited set of model interaction for users according to predetermined logic. In Mithril and other MVC frameworks you are writing the entire view, all possible user interactions, and their effect on the model.

[Here's an example](http://jsbin.com/judibe/edit?js,output) of a pattern for saving the state of objects' property assignments and reverting them at will to mimic native form behaviour.

**Comment by futurist** [comment]
Thank you for the detailed explain. I see, it's the direction of MVC. Will learn more from example.
